tests/web3_internal/test_wallet.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+import pytest
+
+from ocean_lib.web3_internal.utils import add_ethereum_prefix_and_hash_msg
+from ocean_lib.web3_internal.wallet import Wallet
+from ocean_lib.web3_internal.web3_provider import Web3Provider
+
+
+def test_wallet_arguments():
+    web3 = Web3Provider.get_web3()
+
+    private_key = os.environ.get("TEST_PRIVATE_KEY1")
+    assert private_key, "envvar TEST_PRIVATE_KEY1 is not set."
+
+    # Create wallet with valid private key
+    wallet = Wallet(web3, private_key=private_key)
+    assert wallet.private_key == private_key
+    assert wallet.address
+    signed_message = wallet.sign(add_ethereum_prefix_and_hash_msg("msg-to-sign"))
+    assert signed_message
+
+    # Create wallet with encrypted key and password
+    password = "darksecret"
```

Comment:
Codacy cries about the hardcoded password but in this context I think we can't do anything about that.

Reply:
agreeed.